systers_portal/community/forms.py

```diff
@@ -9,7 +9,30 @@
 from users.models import SystersUser
 
 
-class CommunityForm(ModelFormWithHelper):
+class AddCommunityForm(ModelFormWithHelper):
+    """ Form to create a new Community by admin. """
+    class Meta:
+        model = Community
+        fields = ('name', 'slug', 'order', 'email', 'mailing_list',
+                  'parent_community', 'website', 'facebook', 'googleplus',
+                  'twitter')
+        helper_class = SubmitCancelFormHelper
+        helper_cancel_href = "{% url 'index' %}"
+
+    def __init__(self, *args, **kwargs):
+        self.admin = kwargs.pop('admin')
+        super(AddCommunityForm, self).__init__(*args, **kwargs)
+
+    def save(self, commit=True):
+        """Override save to add admin to the instance"""
+        instance = super(AddCommunityForm, self).save(commit=False)
+        instance.admin = SystersUser.objects.get(user=self.admin)
+        if commit:
+            instance.save()
+        return instance
+
```

Comment:
Here you would want to use a `User` instance and not an `admin`, could you share what happens when you submit a new community after login as `admin`?

Reply:
After I login as admin and submitting the form, a new community is created whose community admin is the admin who added it.

Reply:
The errors were due to my environment using py3.4. 
Changing this line from 
`instance.admin = SystersUser.objects.get(user=self.admin)`
    to `instance.admin = self.admin` solved the error. I will update this and other PRs. 
Thank you @tapasweni-pathak for the help with reproducing the error.